src/sql/workbench/contrib/editData/browser/editDataEditor.ts

```diff
@@ -619,12 +629,27 @@ export class EditDataEditor extends BaseEditor {
 			} else {
 				this._sash.hide();
 			}
+			if (this._resultsEditorContainer) {
+				this.sashVisibilityUpdate();
+			}
 		}
 
 		this._updateTaskbar(newInput);
 		return this._setNewInput(newInput, options);
 	}
 
+	private sashVisibilityUpdate(): boolean {
```

Comment:
Suggest just calling this updateSashVisibility and not having it return everything, then in toggleQueryPane just call this method separately. There's no reason to have this function control showing/hiding the query pane.